tests/test_app.py

```diff
@@ -8,6 +9,20 @@ def test_init():
     assert app
 
 
+def test_run():
+    app = Application()
+    app.run_internally = mock.MagicMock(return_value=True)
+    assert app.run()
+    assert app.run_internally.called
+
+
+def test_run_with_exception():
```

Comment:
Oh, you cannot raise an exception from lambda (I forgot). In that case, you can do:

```python
def run_internally(*args):
    raise RuntimeError(...)
app.run_internally = run_internally
```